backend/app/api/endpoints/knowledge.py

```
@@ -1509,3 +1549,470 @@ def _update_kb_summary_after_deletion(kb_id: int, user_id: int, user_name: str):
+    Returns basic KB info and current user's permission status.
+    Used when user accesses a share link.
+    """
+    from app.schemas.knowledge import KnowledgeShareInfoResponse
+    from app.services.knowledge.permission_service import (
+        can_access_knowledge_base,
+        get_permission_source,
+        get_user_permission_type,
+    )
+
+    kb = (
+        db.query(Kind)
```

Ruff (0.14.14)

Undefined name `Kind`

(F821)